include/envoy/network/filter.h

```diff
@@ -291,6 +292,11 @@ class ListenerFilterCallbacks {
    */
   virtual envoy::config::core::v3::Metadata& dynamicMetadata() PURE;
   virtual const envoy::config::core::v3::Metadata& dynamicMetadata() const PURE;
+
+  /**
+   * @return Object on which filters can share data on a per-request basis.
+   */
+  virtual StreamInfo::FilterState& filterState() const PURE;
```

Comment:
Sorry just noticed this but it seems odd that a const method returns a non-const object? Is this necessary?

Reply:
`FilterState::SetData` is not const so the return value cannot be `const`. Technically acquiring the reference is also `const` although the caller might decide to change it.

Reply:
I could make it into `virtual StreamInfo::FilterState& filterState() PURE;`

Reply:
+1 on that change, thanks. Otherwise LGTM.

/wait

Reply:
fixed in 5fb510a5598d64837ca624a605c2af901f631ccd